assembl/views/api2/auth.py

```diff
@@ -368,31 +368,30 @@ def reset_password(request):
                     localizer.translate(_("This email does not exist")),
                     code=HTTPNotFound.code)
             else:
-                raise JSONError(
-                    localizer.translate(_(generic_error_message)),
-                    code=HTTPNotFound.code)
+                send_change_password_email(request, user, email, discussion=discussion)
                 logger.error("This email does not exist.")
         if account:
             email = account.email
     else:
         error = localizer.translate(_("Please give an identifier"))
         raise JSONError(error)
     if not email:
-        email = user.get_preferred_email()
+        if user:
+            email = user.get_preferred_email()
+        else:
+            email = None
         if not email:
             if not discussion.preferences['generic_errors']:
                 error = localizer.translate(_("This user has no email"))
             else:
-                error = localizer.translate(_(generic_error_message))
                 logger.error("This user has no email.")
-            raise JSONError(error, code=HTTPPreconditionFailed.code)
+                send_change_password_email(request, user, email, discussion=discussion)
```

Comment:
This doesn't make sense to me to call `send_change_password_email` if this does nothing.
Better `return HTTPOk()` ?

Reply:
Again, it is not just about HTTPOk, the idea is to enter in the `send_change_password_email` and it is this function which decides wether to send an email or not, and in any case it shows the message to the user interface